### DISCUSSION TITLE
not finding anything with multiple (keyword) instance in docs or examples

### DISCUSSION THREAD

**Discussion by jsbimra** [speechly/speechly#157] [General]
How we can add multiple unique instance of speech to text on different fields, on 1 single form? 
finding clicking on first one trigger both the mic for listening

**Comment by aukkonen** [comment]
@jsbimra 
Hello! If I understand your question correctly, you have a form with several fields, and would like to use a separate button for each field to trigger speech-to-text?

Assuming that you are integrating Speechly to a website, this requires using our `browser-client` library directly instead of relying on the Web Components. Please see the [README](https://github.com/speechly/speechly/tree/main/libraries/browser-client) of the `browser-client` for detailed instructions.

In short, you can bind the `startContext()` and `stopContext()` functions to multiple buttons, and based on which button was clicked you can direct the resulting transcript to the correct form component. This logic you must implement as part of the button press and `onSegmentChange()´ callbacks.

When using the React Client, the same can be done by mapping the `toggleRecording()` function to different buttons, more details [here](https://github.com/speechly/speechly/tree/main/libraries/react-client).

**Comment by arzga** [comment]
Just to clarify you can detect keywords from a sentence creating a voice interface configuration in Speechly dashboard https://api.speechly.com/dashboard/. Then in your application you can assign keywords (entities) to different fields based on their entity type.

Example here: https://codepen.io/arzga/pen/XWgBdyL

For React, there is a helper package for filling forms with voice: https://github.com/speechly/speechly/tree/main/libraries/react-voice-forms

If you want to add a "mic button" GUI component to each field, then using `browser-client` like explained above allows for most flexibility. It also uses browser resources in most effectively. For mocking up such interfaces you can, however, use the Web Components like here:

https://codepen.io/arzga/pen/MWoGRdq
